packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -83,10 +93,144 @@ test.describe('`Pie Modal is closed`', () => {
 
+            // Assert
             expect(messages).toHaveLength(1);
         });
     });
 });
 
+test.describe('Pie Modal `isDismissible`', () => {
+    test.describe('when `isDismissible` is truthy', () => {
+        test('should contain a close button so the modal can be dismissed', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act & Assert
+            await expect(component.locator('.c-modal-closeBtn')).toBeVisible();
+        });
+
+        test('should close the modal when the close button is clicked', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act
+            await component.locator('.c-modal-closeBtn').click();
+
+            // Assert
+            await expect(component).not.toBeVisible();
+        });
+
+        test('should close the modal when the backdrop is clicked', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: true,
+                    },
+                },
+            );
+
+            // Act
+            await component.locator('dialog').click();
+
+            // Assert
+            await expect(component).not.toBeVisible();
+        });
+
+        test('should close the modal when the ESC key is triggered', async ({ mount, page }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: false,
+                    },
+                },
+            );
+
+            // Act
+            await page.keyboard.press('Escape');
+
+            // Assert
+            await expect(component).not.toBeVisible();
+        });
+    });
+
+    test.describe('when `isDismissible` is falsey', () => {
+        test('should NOT contain a close button so the modal can not be dismissed', async ({ mount }) => {
+            // Arrange
+            const component = await mount(
+                PieModal,
+                {
+                    props: {
+                        isOpen: true,
+                        isDismissible: false,
+                    },
+                },
+            );
+
+            // Act & Assert
+            await expect(component).not.toBeVisible();
```

Comment:
Is this asserting that the modal isn't visible, rather than the close button?

Reply:
Good spot.